bokehjs/src/lib/core/util/templating.ts

```diff
@@ -78,13 +76,17 @@ function  _get_special_value(name: string, special_vars: Vars) {
     throw new Error(`Unknown special variable '\$${name}'`)
 }
 
-export function _get_column_value(name: string, data_source: ColumnarDataSource, i: Index): unknown | null {
+export function _get_column_value(name: string, data_source: ColumnarDataSource, i: Index | null): unknown | null {
   const column = data_source.get_column(name)
 
   // missing column
   if (column == null)
     return null
 
+  // null index (e.g for patch)
+  if (i == null)
+    return null
```

Comment:
Hits on glyphs like `Patch`, `Area` don't have a sensible index so these functions just need to handle those cases. I used `null` but am open to suggestions for other "not an index"-index values.